R/error-checks.R

```diff
@@ -486,3 +500,71 @@ validateHasUniqueValues <- function(data, dataName = "dataset", na.rm = TRUE, nu
   }
   stop(messages$errorHasNoUniqueValues(data, dataName, na.rm))
 }
+
+validateIsIncludedInDataset <- function(columnNames, dataset, datasetName = NULL, nullAllowed = FALSE, logFolder = NULL) {
+  if (nullAllowed && is.null(columnNames)) {
+    return()
+  }
+  if (isIncluded(columnNames, names(dataset))) {
+    return()
+  }
+  if (is.null(logFolder)) {
+    stop(messages$errorNotIncludedInDataset(columnNames, dataset, datasetName), call. = FALSE, immediate. = TRUE)
+  }
+  logErrorThenStop(messages$errorNotIncludedInDataset(columnNames, dataset, datasetName))
+}
+
+checkIsIncludedInDataset <- function(columnNames, dataset, datasetName = NULL, nullAllowed = FALSE, logFolder = NULL) {
+  if (nullAllowed && is.null(columnNames)) {
+    return()
+  }
+  if (isIncluded(columnNames, names(dataset))) {
+    return()
+  }
+  if (is.null(logFolder)) {
+    warning(messages$errorNotIncludedInDataset(columnNames, dataset, datasetName), call. = FALSE, immediate. = TRUE)
+    return()
+  }
+  logWorkflow(
+    message = messages$errorNotIncludedInDataset(columnNames, dataset, datasetName),
+    pathFolder = logFolder,
+    logTypes = c(LogTypes$Debug, LogTypes$Error)
+  )
+}
+
+validateUnitDataDefinition <- function(unit, unitColumn, observedDataset, outputs = NULL) {
+  # In case, value from reading from Excel/csv file is not an actual NULL
+  if (any(isOfLength(unit, 0), is.na(unit), unit %in% "")) {
+    unit <- NULL
+  }
+  # Case unit is defined using outputs
+  dataUnit <- NULL
+  if (!isOfLength(outputs, 0)) {
+    dataUnit <- unlist(lapply(outputs, function(output) {
+      output$dataUnit
+    }))
+  }
+
+  # Checks for errors
+  # If no unit defined at all
+  if (isOfLength(c(unit, unitColumn, dataUnit), 0)) {
+    stop(messages$errorNoDataUnit())
+  }
+  # If no unit defined by dictionray, all outputs need to define dataUnit
+  if (isOfLength(c(unit, unitColumn), 0)) {
+    if (!isSameLength(dataUnit, outputs)) {
+      stop(messages$errorNoDataUnitInOutputs())
+    }
+    return(invisible())

```

Comment:
sometime you return `return()` sometime `return(invisible())`
What's the difference?

Reply:
The difference is the print of the return.
Adding `invisible` just enforce that `NULL` is not printed on the console whenever a `validate` method is used.

In more details, if you have these 2 functions: `f1 <- function(x){x}` and `f2 <- function(x){invisible(x)}`
Then, f1 prints what you return but not f2.
```
>f1(NULL)
NULL
>f1("a")
[1] "a"
>x <- f1("a") # when allocated, the output is not printed anyways
```
while
```
>f2(NULL)
>f2("a")
```

Reply:
ok so in the validate method, it should always be return(invisible()) correct? This is not the case I believe